src/USER-BOCS/fix_bocs.cpp

```diff
@@ -807,21 +856,35 @@ void FixBocs::build_cubic_splines( double **data )
 
     d[j] = (c[j+1]-c[j])/(3.0 * h[j]);
   }
-  splines = (double **) calloc(5,sizeof(double *));
 
-  for ( idx = 0; idx < 5; ++idx)
-  {
-    splines[idx] = (double *) calloc(n-1,sizeof(double));
-  }
-  idx = 0;
-  for ( idx = 0; idx < n - 1; ++idx)
+  memory->create(splines, NUM_CUBIC_SPLINE_COLUMNS, n-1, "splines");
+  int numSplines = 0;
+  for (int idx = 0; idx < n - 1; ++idx)
   {
+    splines[0][idx] = data[0][idx];
     splines[1][idx] = a[idx];
     splines[2][idx] = b[idx];
     splines[3][idx] = c[idx];
     splines[4][idx] = d[idx];
-    splines[0][idx] = data[0][idx];
+    numSplines++;
```

Comment:
why count here? numSplines would be n-1.

Reply:
Count was vestigial, intended to be a sort of a breadcrumb for original authors to orient themselves as they compared old/new code. But you're absolutely right; computationally it makes no sense. Code simplified now.